index.js

```diff
@@ -1,79 +1,99 @@
+            return callback(null, matchups);
+        });
+    }
 
+    static trimWhiteSpace(arr) {
 
-internals.Probables.get = function (day, callback) {
+        return arr.map((elem) => {
 
-    const url = `http://mlb.mlb.com/news/probable_pitchers/index.jsp?c_id=mlb&date=${day}`;
-    const scope = 'div#mc';
-    const selector = {
-        pitchers: ['div.pitcher@pid'],
-        names: ['div.pitcher h5 a'],
-        throws: ['div.pitcher h5 span'],
-        teams: ['div.pitcher@tid'],
-        games: ['div.pitcher@gid'],
-        startTimes: ['div.pitcher@local_time'],
-        easternTimes: ['div.pitcher@eastern_time'],
-        timezones: ['div.pitcher@local_time_zone']
-    };
+            return elem.replace(/\s/g, '');
+        });
+    }
 
-    const x = Xray();
-    x(url, scope, selector)((err, result) => {
+    static convertResult(result) {
 
-        if (err) {
-            return callback(err);
-        }
+        const matchups = [];
+
+        const pitcherIDInURL = /.*(\d{6}).*/;
+        const pitcherIDs = result.pitchers.map((pitcher) => {
 
-        const matchups = internals.convertResult(result);
-        return callback(null, matchups);
-    });
-};
-
-
-internals.convertResult = function (result) {
-
-    const matchups = [];
-
-    const pitchers = result.pitchers;
-    const teams = result.teams;
-    const names = result.names;
-    const throws = result.throws;
-
-    for (let i = 0; i < pitchers.length; i += 2) {
-        const j = i + 1;
-        const matchup = {
-            id: result.games[i],
-            startTime: result.startTimes[i],
-            easternTime: result.easternTimes[i],
-            timezone: result.timezones[i],
-            teams: {
-                away: teams[i],
-                home: teams[j]
-            },
-            pitchers: {
-                away: {
-                    id: pitchers[i],
-                    name: names[i],
-                    throws: throws[i]
+            const matches = pitcherIDInURL.exec(pitcher);
+            if ( matches.length !== 2 ) {
+                throw `Unexpected URL format while trying to extract pitcher ID: ${pitcher}`;
+            }
+            return matches[1];
+        });
+        const teams = Probables.trimWhiteSpace(result.teams);
+        const names = result.names;
+        const throws = Probables.trimWhiteSpace(result.throws);
+
+        for (let i = 0; i < pitcherIDs.length; i += 2) {
+            const j = i + 1;
+            const matchup = {
+                id: result.games[i],
+                startTime: result.startTimes[i],
+                easternTime: result.easternTimes[i],
+                timezone: result.timezones[i],
+                teams: {
+                    away: teams[i],
+                    home: teams[j]
                 },
-                home: {
-                    id: pitchers[j],
-                    name: names[j],
-                    throws: throws[j]
+                pitchers: {
```

Comment:
This technically makes this a breaking change, but it makes sense

Reply:
sorry – which part breaks it? I'd intended to keep it backward compatible

Reply:
To be clear, I don't think the JSON format has changed at all – please let me know if it has.

Reply:
The object originally only had `home` and `away` properties that had the pitcher details.  Now those two objects are moved into a `pitchers` property.  I don't hate the change, it makes sense